spynnaker/pyNN/extra_algorithms/splitter_components/splitter_abstract_pop_vertex_neurons_synapses.py

```diff
@@ -483,9 +483,16 @@ def __add_plastic_feedback(
         if (app_vertex.synapse_dynamics is not None and
                 not isinstance(app_vertex.synapse_dynamics,
                                SynapseDynamicsStatic)):
-            neuron_to_synapse_edge = MachineEdge(neuron_vertex, synapse_vertex)
-            machine_graph.add_edge(neuron_to_synapse_edge, SPIKE_PARTITION_ID)
-            synapse_vertex.set_neuron_to_synapse_edge(neuron_to_synapse_edge)
+            if (app_vertex.self_projection is None):
+                neuron_to_synapse_edge = MachineEdge(
+                    neuron_vertex, synapse_vertex)
+                machine_graph.add_edge(
+                    neuron_to_synapse_edge, SPIKE_PARTITION_ID)
+                synapse_vertex.set_neuron_to_synapse_edge(
+                    neuron_to_synapse_edge)
+            else:
+                synapse_vertex.set_neuron_vertex_and_partition_id(
+                    neuron_vertex, SPIKE_PARTITION_ID)
```

Comment:
This is fine, but note that in both cases you could use this call since you always know the neuron vertex and SPIKE_PARTITION_ID.  That would then save an "if" in the destination.